inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/reader/external/IncrementalSourceStreamFetcher.java

```diff
@@ -164,8 +171,7 @@ private boolean shouldEmit(SourceRecord sourceRecord) {
         if (taskContext.isDataChangeRecord(sourceRecord)) {
             TableId tableId = taskContext.getTableId(sourceRecord);
             Offset position = taskContext.getStreamOffset(sourceRecord);
-            // source record has no primary need no comparing for binlog position
-            if (hasEnterPureStreamPhase(tableId, position) || sourceRecord.key() == null) {
+            if (hasEnterPureStreamPhase(tableId, position)) {
```

Comment:
These changes will affect this PR https://github.com/apache/inlong/pull/7489, so you should not modify it.